sdk/core/core/CMakeLists.txt

```diff
@@ -8,7 +8,7 @@ project (az_core LANGUAGES C)
 set(CMAKE_C_STANDARD 99)
 
 add_library (
-  az_core STATIC
```

Comment:
Let's keep static?

Reply:
Can you help me understand the difference and when we have them as static vs. not?

Reply:
Explained offline :)
We don't want to block people to build shared libs if they want
see: https://github.com/Azure/azure-sdk-for-c/issues/358